src/SFA.DAS.EAS.AcceptanceTests.Common/Extensions/ContainerExtensions.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System.Data;
+using HMRC.ESFA.Levy.Api.Client;
+using SFA.DAS.Authentication;
+using SFA.DAS.Authorization;
+using SFA.DAS.EAS.AcceptanceTests.Common.Steps;
+using SFA.DAS.EAS.Domain.Interfaces;
+using SFA.DAS.EAS.Domain.Models.Account;
+using SFA.DAS.EAS.Web.ViewModels;
+using SFA.DAS.Events.Api.Client;
+using StructureMap;
+
+namespace SFA.DAS.EAS.AcceptanceTests.Common.Extensions
+{
+    public static class ContainerExtensions
+    {
+
+        public static IContainer FinalizeTransactions(this IContainer container)
+        {
+            container.GetInstance<SFA.DAS.EAS.Infrastructure.Data.EmployerAccountsDbContext>().Database.CurrentTransaction
+                .Rollback();
+            container.GetInstance<SFA.DAS.EAS.Infrastructure.Data.EmployerFinanceDbContext>().Database.CurrentTransaction
+                .Rollback();
+
+            return container;
+        }
+
+        public static IContainer InitialiseAndUseInfrastructureDbContexts(this IContainer container)
+        {
+            container
+                .InitialiseAndUseInfrastructureEmployerAccountsDbContext()
+                .InitialiseAndUseInfrastructureEmployerFinanceDbContext();
+
+            return container;
+        }
+
+        public static IContainer UseObjectContextProperties(this IContainer container, ObjectContext objectContext)
+        {
+            container.Configure(c =>
+            {
+                c.For<IApprenticeshipLevyApiClient>().Use(objectContext.ApprenticeshipLevyApiClient.Object);
+                c.For<IAuthenticationService>().Use(objectContext.AuthenticationService.Object);
+                c.For<IAuthorizationService>().Use(objectContext.AuthorizationService.Object);
+                c.For<ICookieStorageService<EmployerAccountData>>().Use(objectContext.CookieEmployerAccountDataStorageService.Object);
+                c.For<ICookieStorageService<FlashMessageViewModel>>().Use(objectContext.CookieFlashMessageViewModelStorageService.Object);
+                c.For<ICurrentDateTime>().Use(objectContext.CurrentDateTimeMock.Object);
+                c.For<IEventsApi>().Use(objectContext.EventsApiMock.Object);
+            });
+
+            return container;
+        }
+
+        public static IContainer InitialiseAndUseInfrastructureEmployerAccountsDbContext(this IContainer container)
+        {
+            var infrasctuctureEmployerAccountsDbContext = container.GetInstance<SFA.DAS.EAS.Infrastructure.Data.EmployerAccountsDbContext>();
+            infrasctuctureEmployerAccountsDbContext.Database.BeginTransaction(IsolationLevel.ReadUncommitted);
```

Comment:
Eventually only one DB context will be needed to be initialized but since the decoupling is still ongoing unfortunately wiring up multiple is necessary. Instead of resolving the DB context, beginning a transaction, re-configuring the container etc though you could register multiple instances of `IUnitOfWorkManager` in the container in the `EmployerAccountsAcceptanceTestsRegistry` and call `Begin()` & `End()` on them before and after a controller/orchestrator method is called e.g. `_container.GetInstance<IEnumerable<IUnitOfWorkManager>>().ToList().ForEach(m => m.Begin());`